### ISSUE TITLE
Enhancements to "individual driving style" feautre

### ISSUE THREAD

**Issue by aubergine10** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#339]
As per #334 I'm splitting out the additional ideas that were out of scope of that PR.

The PR introduces the notion of 'individual driving styles', primarily as a means to overcome issues with DLS-induced lane-bunching as described in #263. Specifically, each cim gets a deterministic pseudo-random seed that alters when they will make lane changes, thus preventing the lane bunching effect caused by the previous implementation.

I would like to see if we could extend the driving style functionality in any of the following ways...

## Environmental

Lane jumping and speed should be affected by:

* Road maintenance condition (I think this already happens)
* Rain/snow intensity
* Fog intensity
* Temperature (particularly below 4 deg C)
* Wind intensity
* Sun light intensity

For example, if there is dense fog on a frosty night, people will be driving real slow and careful, and not doing much if any lane jumping because they can't see far enough ahead.

Wind intensity could be calculated per road segment, or per vehicle grid cell, or whatever is fast, rather than specific map positions. It doesn't need to be super accurate. It also doesn't need to update every frame - none of the stuff above does.

## Cims

Lane jumping and speed should be affected by:

* Age of cim
* Whether they are heading to work or not

## Traffic / Roads

* More likely to overtake on bigger roads, order of preference:
    * Highway
    * Large roads
    * Medium roads
    * Small roads
    * Note: Might need to factor in the extra categories that NExt2 adds
* More likely to overtake if leader car is going slow

**Comment by RenaKunisaki** [comment]
Idea: cims driving home from work are tired -> they drive slower and less aggressively?

**Comment by aubergine10** [comment]
Yup, but would checking "is this cim driving home" for 16k vehicles (64k with the recent More Vehicles mod) going to grind game to a halt? Hence I was thinking we could just approximate it based on time of day / daylight / weather etc.

**Comment by VictorPhilipp** [comment]
We already know that a cim returns home from "something" (work/shopping/visitting/etc.).
